### PR TITLE
CONSOLE-2372: i18n - Networking pages

### PR DESCRIPTION
**Pull request by dtaylor113**
Administrator -> Networking:
- Services
- Routes
- Ingresses
- Network  Policies

### CODE REVIEW

File: frontend/public/components/routes.tsx [hunk 1]
```diff
@@ -233,25 +195,30 @@ const RouteTableRow: RowFunction<RouteKind> = ({ obj: route, index, key, style }
...
-      <DetailsItem label="Certificate" obj={route} path="spec.tls.certificate">
+      <DetailsItem label={t('network-route~Certificate')} obj={route} path="spec.tls.certificate">
         {tls.certificate ? <CopyToClipboard value={tls.certificate} /> : '-'}
       </DetailsItem>
       <dt className="co-m-route-tls-reveal__title">
-        Key{' '}
+        {t('network-route~Key')}{' '}
         {tls.key && (
           <Button
             className="pf-m-link--align-left"
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:224]: Hide and Reveal text below needs to be externalized.

File: frontend/public/components/routes.tsx [hunk 2]
```diff
@@ -353,69 +324,82 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
+            >
+              {ingress.wildcardPolicy}
+            </DetailsItem>
+            <DetailsItem
+              label={t('network-route~Router canonical hostname:')}
+              obj={route}
+              path="status.ingress.routerCanonicalHostname"
+            >
+              <div>{ingress.routerCanonicalHostname || '-'}</div>
+              {true && (
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:376]: Don't forget to revert this change for testing :)

File: frontend/public/components/service.jsx [hunk 3]
```diff
@@ -142,121 +103,139 @@ const ServiceAddress = ({ s }) => {
     switch (type) {
       case 'NodePort':
         return ServiceIPsRow(
-          'Node Port',
-          'Accessible outside the cluster',
+          t('network-service~Node Port'),
```
Comment (spadgett) [inline frontend/public/components/service.jsx:106]:
```suggestion
          t('network-service~Node port'),
```

File: frontend/public/components/service.jsx [hunk 4]
```diff
@@ -142,121 +103,139 @@ const ServiceAddress = ({ s }) => {
...
+          t('network-service~Accessible outside the cluster'),
           _.map(s.spec.ports, 'nodePort'),
-          '(all nodes): ',
+          t('network-service~(all nodes): '),
         );
       case 'LoadBalancer':
         return ServiceIPsRow(
-          'External Load Balancer',
-          'Ingress point(s) of load balancer',
+          t('network-service~External Load Balancer'),
```
Comment (spadgett) [inline frontend/public/components/service.jsx:113]:
```suggestion
          t('network-service~External load balancer'),
```

File: frontend/public/components/service.jsx [hunk 5]
```diff
@@ -142,121 +103,139 @@ const ServiceAddress = ({ s }) => {
...
+          t('network-service~Ingress point(s) of load balancer'),
           _.map(s.status.loadBalancer.ingress, (i) => i.hostname || i.ip || '-'),
         );
       case 'ExternalName':
         return ServiceIPsRow(
-          'External Service Name',
-          'Location of the resource that backs the service',
+          t('network-service~External Service Name'),
```
Comment (spadgett) [inline frontend/public/components/service.jsx:119]:
```suggestion
          t('network-service~External service name'),
```

**Review by spadgett — COMMENTED** [review]
Looks good overall, just a few comments. Thanks @dtaylor113 👍

File: frontend/public/components/ingress.jsx [hunk 6]
```diff
@@ -199,47 +169,87 @@ const RulesRows = (props) => {
...
-    </div>
-  </>
-);
-
+      <div className="co-m-pane__body">
+        <SectionHeading text={t('ingress~Ingress rules')} />
+        <p className="co-m-pane__explanation">
+          {t(
+            'ingress~rules_description',
+            'These rules are handled by a routing layer (Ingress Controller) which is updated as the rules are modified. The Ingress controller implementation defines how headers and other metadata are forwarded or manipulated',
```
Comment (rebeccaalpert) [inline frontend/public/components/ingress.jsx:190]: I'm concerned using keys and default values will make it difficult to pass files off to translation. (They pretty much only get the keys.) Do you have thoughts on that @spadgett? Thanks.

File: frontend/public/components/network-policy.jsx [hunk 7]
```diff
@@ -178,26 +184,28 @@ const IngressRow = ({ ingress, namespace, podSelector }) => {
...
-        <SectionHeading text="Ingress Rules" />
+        <SectionHeading text={t('network-policy~Ingress rules')} />
         <p className="co-m-pane__explanation">
-          Pods accept all traffic by default. They can be isolated via Network Policies which
-          specify a whitelist of ingress rules. When a Pod is selected by a Network Policy, it will
-          reject all traffic not explicitly allowed via a Network Policy. See more details in{' '}
+          {t(
+            'network-policy~ingress_rules_description',
+            'Pods accept all traffic by default. They can be isolated via Network Policies which specify a whitelist of ingress rules. When a Pod is selected by a Network Policy, it will reject all traffic not explicitly allowed via a Network Policy. See more details in:',
+          )}
```
Comment (rebeccaalpert) [inline frontend/public/components/network-policy.jsx:200]: I'm concerned about defined keys here as well.

File: frontend/public/components/routes.tsx [hunk 8]
```diff
@@ -353,69 +324,82 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
+      headerContent={<>{t('network-route~Custom route')}</>}
+      bodyContent={
+        <div>
+          <p>
+            <Trans i18nKey="network-route~custom_route">
+              To use a custom route, you must update your DNS provider by creating a canonical name
+              (CNAME) record. Your CNAME record should point to your custom domain{' '}
+              <strong>{{ host }}</strong>, to the OpenShift canonical router hostname,{' '}
+              <strong>{{ routerCanonicalHostname }}</strong>, as the alias.
+            </Trans>
```
Comment (rebeccaalpert) [inline frontend/public/components/routes.tsx:340]: Worried about keys here too. If you do something like `<Trans t={t} ns="network-route">` you can use the text as the key.

**Review by rebeccaalpert — APPROVED** [review]
Looks good to me aside from the capitalization bits Sam noted. I'm just concerned about the impact of using keys instead of text on translation given that the translators only really see the keys. @spadgett really makes the final decision on that though.

File: frontend/public/components/routes.tsx (hunk 2 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx:376]: yikes!   -thanks :-)

File: frontend/public/components/ingress.jsx (hunk 6 above)
Comment (dtaylor113) [reply frontend/public/components/ingress.jsx:190]: I thought this was neccesary due to some character limit of the keys?

Comment (rebeccaalpert) [reply frontend/public/components/ingress.jsx:190]: I believe that was actually an issue with i18n not being configured to allow periods. We had the parser configured, but not i18n itself. I think Robb said it was resolved.

Comment (rebeccaalpert) [reply frontend/public/components/ingress.jsx:190]: Let me know if you run into something weird like that though.

File: frontend/public/components/network-policy.jsx (hunk 7 above)
Comment (dtaylor113) [reply frontend/public/components/network-policy.jsx:200]: @spadgett & @rebeccaalpert, I think we need a ruling here :-)  I'm concerned other devs might be using keys.  We had a thread about their usage, and I thought it was best practice to use the key for long descriptions.

Comment (dtaylor113) [reply frontend/public/components/network-policy.jsx:200]: I see some keys being used in @rhamilto 's PR [here](https://github.com/openshift/console/pull/6883/files#diff-938bd1f94fe7531ac8316dbfa825e7947b180102cdbd171e53c8758af149030fR20).

Comment (rebeccaalpert) [reply frontend/public/components/network-policy.jsx:200]: Yeah, I haven't reviewed Robb's yet. It's up to Sam ultimately; I'm just concerned about the impact on translation.

File: frontend/public/components/ingress.jsx (hunk 6 above)
Comment (spadgett) [reply frontend/public/components/ingress.jsx:190]: Originally we had a problem with periods, but I thought that was resolved. @rhamilto, did you switch to using the full string as a key?

The translators should have access to the English file, so I think they could figure it out. But I might just use the full message as the key everywhere so we're consistent and there's no confusion.

**Comment by dtaylor113** [comment]
Hi @spadgett & @rebeccaalpert, I believe I fixed all the capitalization issues.

There is still a question about [defined keys](https://github.com/openshift/console/pull/6928#discussion_r504879052).

And I am also seeing this issue using this simple parameter subsitution:
```
t('network-policy~All traffic is allowed to Pods in {{namespace}}', {
  namespace: np.metadata.namespace,
})
```
which produces:
"All traffic is allowed to Pods in {{namespace}}": "All traffic is allowed to Pods in {{namespace}}"

English output seems fine:
"All traffic is allowed to Pods in daves-project"

but pseudo-translation is translating the variable values!?
[ÀÀḽḽ ţṛααϝϝḭḭͼ ḭḭṡ ααḽḽṓṓẁḛḛḍ ţṓṓ Ṕṓṓḍṡ ḭḭṇ ḍααṽḛḛṡ-ṗṛṓṓĵḛḛͼţ]
Shouldn't this be:
[ÀÀḽḽ ţṛααϝϝḭḭͼ ḭḭṡ ααḽḽṓṓẁḛḛḍ ţṓṓ Ṕṓṓḍṡ ḭḭṇ daves-project]  ?

Other parameter subs are working, but they are using the `<Trans />` component though

**Comment by rebeccaalpert** [comment]
@dtaylor113 - Sam made a ruling here: https://github.com/openshift/console/pull/6928#discussion_r504948558

Unfortunately, that's an issue with the pseudolocalization library we're using. It's pretty bare-bones as far as code goes. Maybe when this project is over we can look into contributing and fixing that bug.

File: frontend/public/components/routes.tsx [hunk 9]
```diff
@@ -512,9 +500,56 @@ export const RoutesDetailsPage: React.FC<RoutesDetailsPageProps> = (props) => (
...
+      {
+        title: '',
+        props: { className: tableColumnClasses[5] },
+      },
+    ];
+  };
+  return (
+    <Table
+      {...props}
+      aria-label="Routes"
```
Comment (rebeccaalpert) [inline frontend/public/components/routes.tsx:546]: This aria label should be marked for translation.

**Review by rebeccaalpert — COMMENTED** [review]
Capitalization looks good to me!

File: frontend/public/components/ingress.jsx (hunk 6 above)
Comment (dtaylor113) [reply frontend/public/components/ingress.jsx]: Changed to use full message as the key.

File: frontend/public/components/network-policy.jsx (hunk 7 above)
Comment (dtaylor113) [reply frontend/public/components/network-policy.jsx:199]: Changed to use full message as the key.

File: frontend/public/components/routes.tsx (hunk 8 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx:340]: Changed to use `<Trans />` element as suggested, thanks!

File: frontend/public/components/routes.tsx (hunk 9 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx]: fixed, good catch, thought I did a search on those files, just redid one, no untranslated `aria-label`'s found in these files.

**Comment by dtaylor113** [comment]
Hi @spadgett & @rebeccaalpert, I believe I have addressed all comments in this PR.  PTAL  -thanks :-)

**Review by rebeccaalpert — APPROVED** [review]
Looks good to me! Thank you.

File: frontend/public/components/routes.tsx (hunk 9 above)
Comment (rebeccaalpert) [reply frontend/public/components/routes.tsx]: I missed it my first pass! :) I swear they hide.

File: frontend/public/components/routes.tsx [hunk 10]
```diff
@@ -353,69 +324,82 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
-);
+const RouteIngressStatus: React.FC<RouteIngressStatusProps> = ({ route }) => {
+  const { t } = useTranslation();
+  return (
+    <>
+      {_.map(route.status.ingress, (ingress: RouteIngress) => (
+        <div key={ingress.routerName} className="co-m-route-ingress-status">
+          <SectionHeading text={`${t('network-route~Router:')} ${ingress.routerName}`} />
+          <dl>
+            <DetailsItem label={t('network-route~Host:')} obj={route} path="status.ingress.host">
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:360]: We shouldn't include a `:` in the property names passed to `DetailsItem`

File: frontend/public/components/routes.tsx [hunk 11]
```diff
@@ -353,69 +324,82 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
-    ))}
-  </>
-);
+const RouteIngressStatus: React.FC<RouteIngressStatusProps> = ({ route }) => {
+  const { t } = useTranslation();
+  return (
+    <>
+      {_.map(route.status.ingress, (ingress: RouteIngress) => (
+        <div key={ingress.routerName} className="co-m-route-ingress-status">
+          <SectionHeading text={`${t('network-route~Router:')} ${ingress.routerName}`} />
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:358]: Ideally `routerName` is a parameter in the message

File: frontend/public/components/service.jsx [hunk 12]
```diff
@@ -142,121 +103,139 @@ const ServiceAddress = ({ s }) => {
...
           _.map(s.spec.ports, 'nodePort'),
-          '(all nodes): ',
+          t('network-service~(all nodes): '),
         );
       case 'LoadBalancer':
         return ServiceIPsRow(
-          'External Load Balancer',
-          'Ingress point(s) of load balancer',
+          t('network-service~External load balancer'),
+          t('network-service~Ingress point(s) of load balancer'),
```
Comment (spadgett) [inline frontend/public/components/service.jsx:114]: nit: Not new, but I would remove the parentheses around the `s`

File: frontend/public/components/service.jsx [hunk 13]
```diff
@@ -142,121 +103,139 @@ const ServiceAddress = ({ s }) => {
...
+        <div className="col-xs-6">{t('network-service~Location')}</div>
       </div>
       <div className="rows">
         {ServiceType(s.spec.type)}
         {s.spec.externalIPs &&
           ServiceIPsRow(
-            'External IP',
-            'IP Address(es) accepting traffic for service',
+            t('network-service~External IP'),
+            t('network-service~IP Address(es) accepting traffic for service'),
```
Comment (spadgett) [inline frontend/public/components/service.jsx:143]: I'd remove the parentheses here too

File: frontend/public/components/routes.tsx (hunk 10 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx]: fixed, and also the two below this one :-)

File: frontend/public/components/routes.tsx (hunk 11 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx]: fixed

File: frontend/public/components/service.jsx (hunk 12 above)
Comment (dtaylor113) [reply frontend/public/components/service.jsx]: fixed

File: frontend/public/components/service.jsx (hunk 13 above)
Comment (dtaylor113) [reply frontend/public/components/service.jsx]: fixed

File: frontend/public/components/routes.tsx (hunk 11 above)
Comment (spadgett) [reply frontend/public/components/routes.tsx]: `:` is not a problem in the key. It's simply that we don't use `:` in the `DetailsItem` labels. It's fine here in the header.

File: frontend/public/components/routes.tsx [hunk 14]
```diff
@@ -353,69 +324,90 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
+      {_.map(route.status.ingress, (ingress: RouteIngress) => (
+        <div key={ingress.routerName} className="co-m-route-ingress-status">
+          <SectionHeading
+            text={`${t('network-route~Router: {{routerName}}', {
+              routerName: ingress.routerName,
+            })}`}
+          />
+          <dl>
+            <DetailsItem
+              label={`${t('network-route~Host')}:`}
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:365]:
```suggestion
              label={t('network-route~Host')}
```

File: frontend/public/components/routes.tsx [hunk 15]
```diff
@@ -353,69 +324,90 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
+              obj={route}
+              path="status.ingress.host"
+            >
+              {ingress.host}
+            </DetailsItem>
+            <DetailsItem
+              label={`${t('network-route~Wildcard policy')}:`}
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:372]:
```suggestion
              label={t('network-route~Wildcard policy')}
```

File: frontend/public/components/routes.tsx [hunk 16]
```diff
@@ -353,69 +324,90 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
+              obj={route}
+              path="status.ingress.wildcardPolicy"
+            >
+              {ingress.wildcardPolicy}
+            </DetailsItem>
+            <DetailsItem
+              label={`${t('network-route~Router canonical hostname')}:`}
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:379]:
```suggestion
              label={t('network-route~Router canonical hostname')}
```

File: frontend/public/components/routes.tsx (hunk 14 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx:365]: @spadgett What do we do with the `:` ?

File: frontend/public/components/routes.tsx (hunk 16 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx:379]: Are we dropping the `:`?  I guess so, I don't see them on other detail pages, wonder why there are in this one?

File: frontend/public/components/routes.tsx (hunk 11 above)
Comment (dtaylor113) [reply frontend/public/components/routes.tsx]: ah, k , got it -thanks

**Comment by dtaylor113** [comment]
@spadgett, I believe I have addressed all review comments -thanks

**Comment by sg00dwin** [comment]
I noticed that the Routes status filter dropdown label and selections are not translated. Should they be? https://github.com/openshift/console/blob/988aa8f24a9e7e2c223d8fecf29be3a91c0e458e/frontend/public/components/routes.tsx#L558-L569

I have a similar status filter on the VolumeSnapshots list page.

**Comment by dtaylor113** [comment]
> I noticed that the Routes status filter dropdown label and selections are not translated...I have a similar status filter on the VolumeSnapshots list page

@sg00dwin & @rebeccaalpert, this [commit](https://github.com/openshift/console/pull/6928/commits/ddecfc84b26e7aef2fc57b780609d5c38fbab6a8) seemed to work for me:
image

File: frontend/public/components/routes.tsx [hunk 17]
```diff
@@ -353,69 +324,86 @@ const RouteTargetRow: React.FC<RouteTargetRowProps> = ({ route, target }) => (
...
     route.spec.host,
     route.status.ingress,
   );
   return (
     <>
       <div className="co-m-pane__body">
-        <SectionHeading text="Route Details" />
-        <RouteCharts namespace={route.metadata.namespace} route={route.metadata.name} />
+        <SectionHeading text={t('network-route~Route details')} />
+        <RouteCharts namespace={route.metadata.namespace} route={route.metadata.name} t={t} />
```
Comment (spadgett) [inline frontend/public/components/routes.tsx:406]: We shouldn't be passing `t` to components. They should use their own `useTranslation`.

We'll eventually be passing a namespace when we call `useTranslation` so not all `t` are the same.

Comment (dtaylor113) [reply frontend/public/components/routes.tsx]: ok, fixed

File: frontend/public/components/nav/admin-nav.tsx [hunk 18]
```diff
@@ -197,7 +197,7 @@ const AdminNav = () => {
           activePath="/brokermanagement/"
           startsWith={brokerManagementStartsWith}
         />
-      </NavSection>
+      </NavSection> conf
```
Comment (spadgett) [inline frontend/public/components/nav/admin-nav.tsx:200]: Typo?

Comment (dtaylor113) [reply frontend/public/components/nav/admin-nav.tsx]: yup, fixed, thanks

**Review by spadgett — APPROVED** [review]
/lgtm
/hold

**Comment by spadgett** [comment]
/assign @yapei @ahardin-rh @sferich888 
for approvals

File: frontend/public/components/network-policy.jsx [hunk 19]
```diff
@@ -178,26 +184,29 @@ const IngressRow = ({ ingress, namespace, podSelector }) => {
 };
 
 const Details_ = ({ obj: np, flags }) => {
+  const { t } = useTranslation();
   return (
     <>
       <div className="co-m-pane__body">
-        <SectionHeading text="Namespace Details" />
+        <SectionHeading text={t('network-policy~Namespace details')} />
```
Comment (yanpzhan) [inline frontend/public/components/network-policy.jsx:191]: Hi, when checked on network policy detail page, found this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1891419 , so the translation text also need update accordingly.

Comment (dtaylor113) [reply frontend/public/components/network-policy.jsx]: Hi @yanpzhan, PTAL and retest with new 'Network policy details' title (per UXD and I18n aggrement on sentance-case headings).  -thanks

Comment (spadgett) [reply frontend/public/components/network-policy.jsx]: Hi, @yanpzhan can you retest? The heading should be fixed now.

Comment (yanpzhan) [reply frontend/public/components/network-policy.jsx]: Sorry for late response, I'm on leave yesterday. I checked on console against the pr,  the network policy detail page has been fixed.

**Review by spadgett — APPROVED** [review]
/lgtm

Thanks @dtaylor113 👍

**Pull request merged**